client/widgets/Buttons.cpp

```diff
@@ -447,26 +447,89 @@ void CToggleGroup::selectionChanged(int to)
 		parent->redraw();
 }
 
-void CToggleGroup::show(SDL_Surface * to)
+CVolumeSlider::CVolumeSlider(Point position, const std::string &defName, int value,
+                             const std::pair<std::string, std::string> * const help) :
+	value(value),
+	anim(new CAnimation(defName)),
+	image(new CAnimImage(anim, 0, 0, position.x, position.y)),
+	helpHandlers(help)
 {
-	if (musicLike)
+	assert(!defName.empty());
+	addUsedEvents(LCLICK | RCLICK | WHEEL);
+	pos.x += position.x;
+	pos.y += position.y;
+	pos.w = (image->pos.w + 1) * anim->size();
+	pos.h = image->pos.h;
+	type |= REDRAW_PARENT;
+	setVolume(value);
+}
+
+CVolumeSlider::~CVolumeSlider()
+{
+	delete image; // Deletes anim too
```

Comment:
use OBJ_CONSTRUCTION_CAPTURING_ALL in constructor to acquire ownership on child objects